### PR TITLE
issue: 2175248 Reduce usage of snprintf()

### PR DESCRIPTION
**Pull request by pasis**
snprintf() is relatively expensive function and makes significant impact
on performance in CPS scenario. Don't build debug strings unless they're
used with respective log level.

Also, don't use snprintf() for hash calculation. Integer arithmetics is
more appropriate.

### CODE REVIEW

**Comment by igor-ivanov** [comment]
bot:retest

File: src/vma/dev/ring_allocation_logic.cpp [hunk 1]
```diff
@@ -182,6 +190,14 @@ bool ring_allocation_logic::should_migrate_ring()
 	return true;
 }
 
+const char* ring_allocation_logic::to_str()
+{
+	if (unlikely(m_str[0] == '\0')) {
+		snprintf(m_str, sizeof(m_str), "[%s=%p]", m_type, m_owner);
```
Comment (igor-ivanov) [inline src/vma/dev/ring_allocation_logic.cpp:196]: Please consider adding return error check if you see any potential issues. Static analysis tools as BlackDuck can report security issues.

Comment (pasis) [reply src/vma/dev/ring_allocation_logic.cpp]: Please, check the last commit. Does it do what you meant?

File: src/vma/dev/net_device_val.cpp [hunk 2]
```diff
@@ -103,26 +103,33 @@ ring_alloc_logic_attr::ring_alloc_logic_attr(const ring_alloc_logic_attr &other)
...
+		/* m_hash * 33 + byte */	\
+		h = (h << 5) + h + (x & 0xff);	\
+		x >>= 8;			\
+	} while (x != 0);			\
+} while (0)
+
+	HASH_ITER(m_ring_alloc_logic);
+	HASH_ITER(m_ring_profile_key);
+	HASH_ITER(m_user_id_key);
+	HASH_ITER(m_mem_desc.iov_base);
```
[the remark this answers is not in the document]
Comment (pasis) [reply src/vma/dev/net_device_val.cpp]: Fixed.

**Comment by igor-ivanov** [comment]
these changes are pushed into master as https://github.com/Mellanox/libvma/commit/83276a19
excluding commit https://github.com/Mellanox/libvma/pull/895/commits/cc88036dbd8a827ef9cb348d10e32c603b09ebbe

**Pull request closed without merging**